media_kit/lib/src/player/native/core/initializer_isolate.dart

```diff
@@ -0,0 +1,190 @@
+  /// Subsequent sent values are [Pointer<mpv_event>].
+  static void _mainloop(SendPort port) async {
+    // [Completer] used to ensure that the last [mpv_event] is NOT reset to [mpv_event_id.MPV_EVENT_NONE] after waiting using [MPV.mpv_wait_event] again in the continuously running while loop.
+    var completer = Completer();
+
+    // Used to recieve the confirmation messages from the main thread about successful receive of the sent event through [SendPort].
+    // Upon confirmation, the [Completer] is completed & we jump to next iteration of the while loop waiting with [MPV.mpv_wait_event].
+    final receiver = ReceivePort();
+
+    // Send the [SendPort] of internal [ReceivePort].
+    port.send(receiver.sendPort);
+
+    // Received data from the [SendPort] for initialization.
+    late Map<String, String> options;
+    late generated.MPV mpv;
+
+    bool disposed = false;
+
+    Pointer<generated.mpv_handle>? handle;
+
+    // * First received value is [Map<String, String>] of options.
+    // * Second received value is [String] of path to [DynamicLibrary].
+    // * Subsequent [bool] values are used to notify the successful interpretation of the sent event.
+    // * [null] value is used to dispose the event loop.
+    receiver.listen(
+      (message) {
+        if (message is Map<String, String>) {
+          options = message;
+        } else if (message is String) {
+          mpv = generated.MPV(DynamicLibrary.open(message));
+          completer.complete();
+        } else if (message is bool) {
+          completer.complete();
+        } else if (message == null) {
+          if (handle != null) {
+            disposed = true;
+            // Break out of last event await.
+            completer.complete();
+            // Break out of the possible [MPV.mpv_wait_event] call.
+            mpv.mpv_wakeup(handle);
+          }
+        }
+      },
+    );
+
+    // Wait for the [Completer] to complete.
+    await completer.future;
+
+    // Creating [mpv_handle].
+    handle ??= mpv.mpv_create();
+
+    // Set custom defined options before [mpv_initialize].
+    for (final entry in options.entries) {
+      final name = entry.key.toNativeUtf8();
+      final value = entry.value.toNativeUtf8();
+      mpv.mpv_set_option_string(handle, name.cast(), value.cast());
+      calloc.free(name);
+      calloc.free(value);
+    }
+
+    // Initializing [mpv_handle].
+    mpv.mpv_initialize(handle);
+
+    // Sending the address of the created [mpv_handle] & the [SendPort] of the [receivePort].
+    // Raw address is sent as [int] since we cannot transfer objects through Native Ports, only primitives.
+    port.send(handle.address);
+
+    // Lookup for events & send to main thread through [SendPort].
+    // Ensuring the successful sending of the last event before moving to next [MPV.mpv_wait_event].
+    while (!disposed) {
+      completer = Completer();
+      final event = mpv.mpv_wait_event(handle, kReleaseMode ? 1 : 0.1);
```

Comment:
Any value >= 0 essentially means polling. It may slow down Dart's event loop. It may be a good idea to keep -1 & address any issues that it brings some other way.

Reply:
I agree, this is not a great solution. It seems to have trouble with waking up. I have an idea of what might be going on, I'll see what can be done.